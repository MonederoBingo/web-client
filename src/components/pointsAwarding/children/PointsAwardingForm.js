import React from "react";
import NumberInput from "../../common/NumberInput";
import TextInput from "../../common/TextInput";

const PointsAwardingForm = (props) => {
  return (
     <div>
       <NumberInput name="phoneNumber" label="Phone #" onChange={props.onChange} placeholder="Phone #"/>
       <NumberInput name="amount" label="Amount" onChange={props.onChange} placeholder="Amount"/>
       <TextInput name="saleKey" label="Sale Key" onChange={props.onChange} placeholder=""/>
       <input id="save" type="submit" disabled={props.saving} value="Save" className="btn btn-primary" onClick={props.onClick} />
     </div>
  );
};

export default PointsAwardingForm;
