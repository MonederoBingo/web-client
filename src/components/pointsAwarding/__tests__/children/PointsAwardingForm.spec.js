import React from 'react';
import expect from 'jest-matchers';
import PointsAwardingForm from '../../children/PointsAwardingForm';
import { shallow } from 'enzyme';
import NumberInput from "../../../common/NumberInput";

describe("PointsAwardingForm", () => {
  it("should show initial layout", () => {
    // given
    const props = {
      onChange: () => {}
    };
     // when
     const wrapper = shallow(<PointsAwardingForm {...props}/>);

     // then
     expect(wrapper.find('div').children().nodes).toMatchSnapshot();
  });
  function fieldShouldCallOnChange(field) {
    // given
    let called = false;
    const props = {
      onChange: () => { called = true; }
    };
    const phoneInput = shallow(<PointsAwardingForm {...props}/>).find("[name='" + field + "']");

     // when
    phoneInput.simulate('change');

     // then
     expect(called).toBe(true);
  }
  it("should call onChange", () => {
     fieldShouldCallOnChange("phoneNumber");
     fieldShouldCallOnChange("amount");
     fieldShouldCallOnChange("saleKey");
  });
});
