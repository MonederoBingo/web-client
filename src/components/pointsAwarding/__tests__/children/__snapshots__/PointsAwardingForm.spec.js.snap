// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsAwardingForm should show initial layout 1`] = `
Array [
  <TextInput
    label="Phone #"
    name="phoneNumber"
    onChange={[Function]}
    placeholder="Phone #"
    value=""
  />,
  <NumberInput
    label="Amount"
    name="amount"
    onChange={[Function]}
    placeholder="Amount"
  />,
  <TextInput
    label="Sale Key"
    name="saleKey"
    onChange={[Function]}
    placeholder=""
  />,
  <input
    className="btn btn-primary"
    disabled={false}
    id="save"
    onClick={[Function]}
    type="submit"
    value="Save"
  />,
]
`;
